services-camel/src/test/java/edu/unc/lib/dl/services/camel/destroyDerivatives/DestroyDerivativesRouterTest.java

```diff
@@ -0,0 +1,109 @@
+/**
+ * Copyright 2008 The University of North Carolina at Chapel Hill
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package edu.unc.lib.dl.services.camel.destroyDerivatives;
+
+import static edu.unc.lib.dl.services.camel.util.CdrFcrepoHeaders.CdrBinaryMimeType;
+import static edu.unc.lib.dl.services.camel.util.CdrFcrepoHeaders.CdrBinaryPidId;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.camel.BeanInject;
+import org.apache.camel.Produce;
+import org.apache.camel.ProducerTemplate;
+import org.apache.camel.builder.AdviceWithRouteBuilder;
+import org.apache.camel.test.spring.CamelSpringTestSupport;
+import org.junit.Test;
+import org.springframework.context.support.AbstractApplicationContext;
+import org.springframework.context.support.ClassPathXmlApplicationContext;
+
+public class DestroyDerivativesRouterTest extends CamelSpringTestSupport {
+    private static final String DESTROY_DERIVATIVES_ROUTE = "CdrDestroyDerivatives";
+    private static final String DESTROY_FULLTEXT_ROUTE = "CdrDestroyFullText";
+    private static final String DESTROY_IMAGE_ROUTE = "CdrDestroyImage";
+
+    @Produce(uri = "direct:start")
+    private ProducerTemplate template;
+
+    @BeanInject(value = "binaryInfoProcessor")
+    private BinaryInfoProcessor binaryInfoProcessor;
+
+    @BeanInject(value = "destroySmallThumbnailProcessor")
+    private DestroyDerivativesProcessor destroySmallThumbnailProcessor;
+
+    @BeanInject(value = "destroyLargeThumbnailProcessor")
+    private DestroyDerivativesProcessor destroyLargeThumbnailProcessor;
+
+    @BeanInject(value = "destroyAccessCopyProcessor")
+    private DestroyDerivativesProcessor destroyAccessCopyProcessor;
+
+    @BeanInject(value = "destroyFulltextProcessor")
+    private DestroyDerivativesProcessor destroyFulltextProcessor;
+
+    @Override
+    protected AbstractApplicationContext createApplicationContext() {
+        return new ClassPathXmlApplicationContext("/service-context.xml", "/destroy-derivatives-context.xml");
+    }
+
+    @Test
+    public void routeRequestText() throws Exception {
+        createContext(DESTROY_DERIVATIVES_ROUTE);
+        template.sendBodyAndHeaders("", createEvent("text/plain"));
+        assertMockEndpointsSatisfied();
```

Comment:
Its not actually making any assertions about the endpoints. Normally it would specify what to expect with something like `getMockEndpoint("mock:direct:solr-update").expectedMessageCount(1);` if you have specify expectations about messages being sent somewhere.